src/redux/actions/samples/saveSamples.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable no-param-reassign */
+import fetchAPI from '../../../utils/fetchAPI';
+import pushNotificationMessage from '../notifications';
+import messages from '../../../components/notification/messages';
+import { SAMPLES_ERROR, SAMPLES_SAVING, SAMPLES_SAVED } from '../../actionTypes/samples';
+
+import errorTypes from './errorTypes';
+
+const saveSamples = (projectUuid) => async (dispatch, getState) => {
+  const project = getState().projects[projectUuid];
+  const { samples } = getState();
+
+  // Get all samples for the project
+
+  const payload = {
+    ids: project.samples,
+  };
+  payload.ids.reduce((acc, sampleUuid) => {
```

Comment:
you can use a filter here. You can also make it a separate function called `getAllSamplesForProject` to make it more readable.